packages/zwave-js/src/lib/commandclass/ThermostatFanModeCC.ts

```diff
@@ -0,0 +1,390 @@
+	mode: ThermostatFanMode;
+};
+
+@CCCommand(ThermostatFanModeCommand.Set)
+export class ThermostatFanModeCCSet extends ThermostatFanModeCC {
+	public constructor(
+		driver: Driver,
+		options:
+			| CommandClassDeserializationOptions
+			| ThermostatFanModeCCSetOptions,
+	) {
+		super(driver, options);
+		if (gotDeserializationOptions(options)) {
+			// TODO: Deserialize payload
+			throw new ZWaveError(
+				`${this.constructor.name}: deserialization not implemented`,
+				ZWaveErrorCodes.Deserialization_NotImplemented,
+			);
+		} else {
+			this.mode = options.mode;
+		}
+	}
+
+	public mode: ThermostatFanMode;
+
+	public serialize(): Buffer {
+		this.payload = Buffer.from([
+			// 0 R R R M M M M
+			this.mode & 0b1111,
+		]);
+		return super.serialize();
+	}
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		const message: MessageRecord = {
+			mode: getEnumMemberName(ThermostatFanMode, this.mode),
+		};
+		return {
+			...super.toLogEntry(),
+			message,
+		};
+	}
+}
+
+@CCCommand(ThermostatFanModeCommand.Report)
+export class ThermostatFanModeCCReport extends ThermostatFanModeCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		validatePayload(this.payload.length >= 1);
+		this._mode = this.payload[0] & 0b1111;
+
+		if (this.version >= 2) {
+			this._off = !!(this.payload[0] & 0b1000_0000);
+		}
+
+		this.persistValues();
+	}
+
+	private _mode: ThermostatFanMode;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.UInt8,
+		min: 0,
+		max: 15,
+		label: "Thermostat fan mode",
+	})
+	public get mode(): ThermostatFanMode {
+		return this._mode;
+	}
+
+	private _off: boolean | undefined;
+	@ccValue({ minVersion: 2 })
+	public get off(): boolean | undefined {
+		return this._off;
+	}
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		const message: MessageRecord = {
+			mode: getEnumMemberName(ThermostatFanMode, this.mode),
+		};
+		if (this.off != undefined) {
+			message.off = this.off;
+		}
+		return {
+			...super.toLogEntry(),
+			message,
+		};
+	}
+}
+
+@CCCommand(ThermostatFanModeCommand.Get)
+@expectedCCResponse(ThermostatFanModeCCReport)
+export class ThermostatFanModeCCGet extends ThermostatFanModeCC {}
+
+@CCCommand(ThermostatFanModeCommand.SupportedReport)
+export class ThermostatFanModeCCSupportedReport extends ThermostatFanModeCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+		this._supportedModes = parseBitMask(
+			this.payload,
+			ThermostatFanMode["Auto high"],
+		);
+
+		// Use this information to create the metadata for the mode property
+		const valueId: ValueID = {
+			commandClass: this.ccId,
+			endpoint: this.endpointIndex,
+			property: "mode",
+		};
+		// Only update the dynamic part
+		this.getValueDB().setMetadata(valueId, ({
+			states: enumValuesToMetadataStates(
+				ThermostatFanMode,
+				this._supportedModes,
+			),
+		} as unknown) as ValueMetadata);
```

Comment:
It is not ideal, but you can get around the double cast by doing this:
```ts
		this.getValueDB().setMetadata(valueId, {
			...ValueMetadata.UInt8, // <== HERE!
			states: enumValuesToMetadataStates(
				ThermostatFanMode,
				this._supportedModes,
			),
		});
```

Reply:
I also copied this bit from ThermostatModeCC.ts. This looks much saner.